test/dotnet-test.UnitTests/GivenThatWeWantToRunTests.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using Microsoft.DotNet.Tools.Test;
+using Microsoft.Extensions.Testing.Abstractions;
+using Moq;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace Microsoft.Dotnet.Tools.Test.Tests
+{
+    public class GivenThatWeWantToRunTests
+    {
+        [Fact]
+        public void Dotnet_test_handles_and_sends_all_the_right_messages()
```

Comment:
Well, there are quite a few of these messages. I believe the test should be the ultimate document of that. Also, the sequence is described here: https://github.com/dotnet/cli/issues/1376